app/controllers/session_controller.rb

```diff
@@ -332,6 +333,34 @@ def create
     end
   end
 
+  def passkey_challenge
+    render json: DiscourseWebauthn.stage_challenge(current_user, secure_session)
+  end
+
+  def passkey_auth_perform
+    raise Discourse::NotFound unless SiteSetting.experimental_passkeys
+
+    params.require(:publicKeyCredential)
+
+    security_key =
+      ::DiscourseWebauthn::AuthenticationService.new(
+        nil,
```

Comment:
I understand why we're passing `nil` to the `current_user` param but wouldn't this blow up in `authenticate_security_key` since `@current_user` will be `nil`?

https://github.com/discourse/discourse/blob/00b1b88a86aa2837e53276c5af805f57f7f183de/lib/discourse_webauthn/authentication_service.rb#L35-L38

Reply:
You may be looking at an outaded version of `authentication_service.rb` (there was a bug here briefly). Now that should check against the key not the current user, because as you note, there is no current user when a passkey is about to be validated. 

https://github.com/discourse/discourse/blob/passkeys-part-2/lib/discourse_webauthn/authentication_service.rb#L35-L38

Reply:
Thank you for clarifying :+1: